src/devices/machine/smc91c9x.cpp

```diff
@@ -441,7 +720,7 @@ READ16_MEMBER( smc91c9x_device::read )
 	}
 
 	if (LOG_ETHERNET && offset != EREG_BANK)
-		logerror("%s:smc91c9x_r(%s) = %04X & %04X\n", machine().describe_context(), ethernet_regname[offset], result, mem_mask);
+		logerror("%s:smc91c9x_r(%s) = %04X & %04X\n", (machine().describe_context()).c_str(), ethernet_regname[offset], result, mem_mask);
```

Comment:
`logerror` is based on `util::stream_format` - it can handle `std::string` fine without `.c_str()` to extract a C string pointer.  Please don't add these in.

Reply:
Done in the new PR. Same root cause as the \r\n problem. Just forgot to revert to the std::string type.

Reply:
For things like this, it's probably less error-prone to define whatever your output function is to behave more like `logerror` than to adjust all call sites to work with your output function.  For example `src/emu/logmacro.h` allows logging to be sent to an a `std::ostream` using the same semantics.  It's possible to use `util::stream_format` with an intermediate `std::ostringstream` or `util::ovectorstream` to do the formatting and then send that unformatted to some other output sink, too.